src/AutoBuild.jl

```diff
@@ -1239,91 +1236,72 @@ function build_jll_package(src_name::String,
+            Core.eval($(src_name)_jll, :(Base.Experimental.@options compile=min optimize=0 infer=false))
+        end
 
-        wrapper_available = false
         \"\"\"
             is_available()
 
         Return whether the artifact is available for the current platform.
         \"\"\"
-        is_available() = wrapper_available
+        function is_available end
+
 
         # We put these inter-JLL-package API values here so that they are always defined, even if there
         # is no underlying wrapper held within this JLL package.
         const PATH_list = String[]
         const LIBPATH_list = String[]
 
-        # We determine, here, at compile-time, whether our JLL package has been dev'ed and overridden
-        override_dir = joinpath(dirname(@__DIR__), "override")
-        if isdir(override_dir)
-            function find_artifact_dir()
-                return override_dir
-            end
-        else
-            function find_artifact_dir()
+        function find_artifact_dir()
+            # We determine at compile-time whether our JLL package has been dev'ed and overridden
+            @static if isdir(joinpath(dirname(@__DIR__), "override"))
+                return joinpath(dirname(@__DIR__), "override")
+            else
                 return artifact"$(src_name)"
             end
+        end
 
-            \"\"\"
-                dev_jll()
-            
-            Check this package out to the dev package directory (usually ~/.julia/dev),
-            copying the artifact over to a local `override` directory, allowing package
-            developers to experiment with a locally-built binary.
-            \"\"\"
-            function dev_jll()
-                # First, `dev` out the package, but don't effect the current project
-                mktempdir() do temp_env
-                    Pkg.activate(temp_env) do
-                        Pkg.develop("$(src_name)_jll")
-                    end
-                end
-                # Create the override directory
-                override_dir = joinpath(Pkg.devdir(), "$(src_name)_jll", "override")
-                # Copy the current artifact contents into that directory
-                if !isdir(override_dir)
-                    cp(artifact"$(src_name)", override_dir)
-                end
-                # Force recompilation of that package, just in case it wasn't dev'ed before
-                touch(joinpath(Pkg.devdir(), "$(src_name)_jll", "src", "$(src_name)_jll.jl"))
-                @info("$(src_name)_ll dev'ed out to $(joinpath(Pkg.devdir(), "$(src_name)_jll")) with pre-populated override directory")
-            end
+        \"\"\"
+            dev_jll()
+        
+        Check this package out to the dev package directory (usually ~/.julia/dev),
+        copying the artifact over to a local `override` directory, allowing package
+        developers to experiment with a locally-built binary.
+        \"\"\"
+        function dev_jll()
```

Comment:
Can't we add something to Pkg or BinaryBuilder to not have to put this into every jll? Like `BinaryBuilder.dev("...")`?

Reply:
https://github.com/JuliaPackaging/JLLWrappers.jl/pull/2#issuecomment-675571311

Reply:
That's loaded by every jll files as well though? It just feels like the wrong place to put it, tools for developing the package shouldn't have to be compiled into the package itself imo. It should be handled by some external thing like `JLLDeveloperTools.jl` or just BinaryBuilder.

Reply:
That's an interesting point of view.

Reply:
I mean, packages don't need an `update_me()` function bundled with them and compiled into their .ji file nor do they need to depend on Pkg to be updatable. Because the process of updating a package is handled externally (and is independent) to the code the package itself. This feels like kinda the same thing. If we want to make the jlls (and the common package they depend on) small, then moving everything dev like that the package won't run itself out to a developer package seems reasonable.

Reply:
To be clear, I think you're probably right that the function doesn't need to be here at all :sweat_smile: